src/app/core/store/collections/collection.converter.ts

```diff
@@ -31,7 +31,22 @@ export class CollectionConverter {
       description: dto.description,
       color: dto.color,
       icon: dto.icon,
-      attributes: dto.attributes ? dto.attributes.map(attr => CollectionConverter.fromAttributeDto(attr)) : [],
+      attributes: dto.attributes ? dto.attributes.map(attr => CollectionConverter.fromAttributeDto(attr))
+        .sort((a, b) => {
+          if (!a.id && !b.id) {
+            return 0;
+          }
+          if (!a.id) {
+            return 1;
+          }
+          if (!b.id) {
+            return -1;
+          }
+
+          const intA = +a.id.substring(1);
+          const intB = +b.id.substring(1);
+          return intA - intB;
+        }) : [],
```

Comment:
I don't know what the future might be. I rather write defensive code. Null or undefined is a valid value of the variables.

Reply:
Sure, it is a good practice to write defensive code. On the other hand, it will just hide the errors from the back-end since ID always needs to be defined.

Reply:
If we should follow some contract (ID must be always set), the contract must be specified somewhere. So let's make a deal, I will turn this into a simpler form and you will specify the contract!

Reply:
Fixed

Reply:
We already have this contract specified in [Document DTO](https://github.com/Lumeer/web-ui/blob/devel/src/app/core/dto/document.ts). The definition `id: string` means that this field should always be set, otherwise it would be `id?: string`.